### DISCUSSION TITLE
How to pass client cert CN to backend as JWT

### DISCUSSION THREAD

**Discussion by hsiung** [apache/apisix#6864] [Q&A]
How do I use APISIX plugins to get the CN of TLS client certificate, extract a substring of it, generate a JWT with this substring as subject and pass the JWT as Authorization header bearer token to the backend?

**Comment by tokers** [comment]
Write your own plugin, using Nginx variables like:

```
$ssl_client_s_dn
```

You can parse the Subject format by yourself.

**Reply by hsiung** [reply]
So does it mean that there are no plugin I can reuse for this quite standard use case? Or does it mean that the plugin concept is flawed (DRY not encouraged)?

**Reply by tzssangglass** [reply]
You can modify or enhance the plugin as needed, which is open source.
